core/shared/src/main/scala-2/org/typelevel/idna4s/core/syntax/CodePointSyntax.scala

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2022 Typelevel
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package org.typelevel.idna4s.core.syntax
+
+import org.typelevel.idna4s.core._
+import org.typelevel.literally.Literally
+
+private[syntax] trait CodePointSyntax {
+  implicit class CodePointContext(val sc: StringContext) {
+    def codePoint(args: Any*): CodePoint = macro CodePointSyntax.codePoint.make
+  }
+}
```

Comment:
Sorry if I missed it, do we have tests for these? You can also test for non-compilation with munit.

https://scalameta.org/munit/docs/assertions.html#compileerrors

Reply:
I don't have tests for those, but I'll add them.